client/core/trade.go

```diff
@@ -37,7 +37,6 @@ func (err ExpirationErr) Error() string { return string(err) }
 // A matchTracker is used to negotiate a match.
 type matchTracker struct {
 	db.MetaMatch
-	id order.MatchID
```

Comment:
But why though? 

`match.id` reads so much better than `match.Match.MatchID`.

Reply:
**TL;DR**
After making the tests track match db updates by match id, I started having issues with cases where `match.id` is set (almost everywhere in the test) and where `match.Match.MatchID` is not set, and since the db update takes in a `MetaMatch`, I could not access and use the `match.id` field. I could have simply bypassed this issue by ensuring that  `match.Match.MatchID` is always set to the same value as `match.id` but looking at the `matchTracker` struct and the embedded `db.MetaMatch` struct, it immediately occurred to me that this duplication of data can be avoided.

I thought about adding a `match.ID()` method for shorter access but saw that there was already an existing `match.ID()` that serves a different purpose.

**Longer version**
`matchTracker` "extends" `db.MetaMatch` and thus has access to all public properties and methods. `db.MetaMatch` already has a `MatchID` field somewhere, it really isn't necessary for `matchTracker` to maintain another property to store same data.

This can all be further simplified by making `db.MetaMatch` embed `MatchMetaData` and `order.UserMatch` rather than as fields:
```
type MetaMatch struct {
	// MatchMetaData is important auxiliary information about the match.
	*MatchMetaData
	// UserMatch is the match info.
	*order.UserMatch
}
```
and access to the common match id can then be `matchTracker.MatchID`. I did not want to make that change in this PR though.

I have a similar field retirement in #957, where I removed the `MatchMetaData.Status` field that's really just a duplicate of the `order.UserMatch.Status` field. `MetaMatch` has references to both `MatchMetaData` and `order.UserMatch` and thus can use the `order.UserMatch.Status` field without requiring an extra `MatchMetaData.Status` field to store the status.

Going with the above suggested change of using embeds instead of fields, the access path may become shorter: `{matchTracker,MetaMatch}.MatchID`, `{matchTracker,MetaMatch}.Status`. I may experiment with that embedding thing in #957 but want to keep unrelated changes here fewer.

Reply:
> but want to keep unrelated changes here fewer

I mean, all of this is an unrelated change, especially since, as you said, 

> I could have simply bypassed this issue by ensuring that `match.Match.MatchID` is always set to the same value as `match.id`

which is exactly what we've done so far. I get wanting to reduce duplication, but this is one where the benefits are worth it for my sanity as a developer. `match.Match.MatchID` is a little too much stuttering for me.

Reply:
True that this is unrelated changes territory. I'm gonna reverse that and do the bypass. I'll probably put up a separate refactor PR that maintains easy readability and reduces duplication and we can discuss whether it's worth making those changes separately.

Reply:
I appreciate that. I think there is a better solution, but as you said, it's out of scope here. 

One thing I'll note is that `(*MetaMatch).ID()` is poorly named. We're only using that as a key for the database, no?

Reply:
I think it's mostly for logging/notifications. Will confirm. I agree it's poorly named; for a long while there, I thought that was the `MatchID` as bytes, only recently recognized it's a combination of match and order ids.

Reply:
I force-pushed out the commit with the unrelated changes and replaced with a more concise fix.